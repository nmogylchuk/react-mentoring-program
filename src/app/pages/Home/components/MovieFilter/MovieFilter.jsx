import React from 'react';
import './MovieFilter.scss';

const MovieFilter = () => {

    return (
        <nav className='movie-filter'>
            <ul className='movie-filter__list'>
                <li className='movie-filter__item'><a href="#" className='movie-filter__link movie-filter__link--active'>All</a></li>
                <li className='movie-filter__item'><a href="#" className='movie-filter__link'>Documentary</a></li>
                <li className='movie-filter__item'><a href="#" className='movie-filter__link'>Comedy</a></li>
                <li className='movie-filter__item'><a href="#" className='movie-filter__link'>Horror</a></li>
                <li className='movie-filter__item'><a href="#" className='movie-filter__link'>Crime</a></li>
            </ul>
        </nav>
    )
};

export default MovieFilter;
