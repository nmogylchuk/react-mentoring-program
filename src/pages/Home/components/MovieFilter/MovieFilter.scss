@import 'src/assets/styles/variables';

.movie-filter {
  &__list {
    display: flex;
    flex-wrap: wrap;
  }

  &__item {
    padding: 20px 15px;
    text-transform: uppercase;
    position: relative;
  }

  &__link {
    &:hover {
      color: $red;
    }

    &:focus, &:active {
      color: $red;
      &:after {
        content: "";
        position: absolute;
        left: 0;
        right: 0;
        border-bottom: 2px solid $red;
        bottom: -14px;
        margin: 10px;
        z-index: 1;
      }
    }
  }

  &__link {
    color: $white;
  }

}
