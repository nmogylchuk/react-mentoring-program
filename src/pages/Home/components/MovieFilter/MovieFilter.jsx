import React from 'react';
import 'pages/Home/components/MovieFilter/MovieFilter.scss';

const MovieFilter = () => {

    let movieFilterData = ['All', 'Documentary', 'Comedy', 'Horror', 'Crime'];

    return (
        <nav className='movie-filter'>
            <ul className='movie-filter__list'>
                {movieFilterData.map((movieFilterItem, index) => (
                    <li key={index} className='movie-filter__item'>
                        <a href="#" className='movie-filter__link'>{movieFilterItem}</a>
                    </li>
                ))}
            </ul>
        </nav>
    )
};

export default MovieFilter;
