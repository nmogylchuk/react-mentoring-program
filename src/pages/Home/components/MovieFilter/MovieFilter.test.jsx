import React from 'react';
import rerender from 'react-test-renderer';
import MovieFilter from './MovieFilter';
import {Provider} from 'react-redux'
import configureStore from 'redux-mock-store';
import {jest} from "@jest/globals";

describe('when MovieFilter', () => {

    test('then snapshot created', () => {
        const initialStore = {};
        const store = configureStore([])(initialStore);
        store.dispatch = jest.fn();
        const movieFilterComponent = rerender.create(<Provider store={store}>
            <MovieFilter/>
        </Provider>);

        const tree = movieFilterComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
