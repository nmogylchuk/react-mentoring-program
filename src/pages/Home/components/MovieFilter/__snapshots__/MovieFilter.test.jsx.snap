// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when MovieFilter then snapshot created 1`] = `
<nav
  className="movie-filter"
>
  <ul
    className="movie-filter__list"
  >
    <button
      onClick={[Function]}
      value="all"
    >
      All
    </button>
    <button
      onClick={[Function]}
      value="documentary"
    >
      Documentary
    </button>
    <button
      onClick={[Function]}
      value="comedy"
    >
      Comedy
    </button>
    <button
      onClick={[Function]}
      value="horror"
    >
      Horror
    </button>
  </ul>
</nav>
`;
